### PR TITLE
Add digamma mathematical function to Paddle frontend

### LINKED ISSUE #18281: digamma

### PR DESCRIPTION
**Pull request by remenraj**
Close #18281

### CODE REVIEW

**Comment by RickSanchezStoic** [comment]
@remenraj could you please use the decorator `@with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")` first  before the other decorators? Thanks!

**Comment by remenraj** [comment]
> @remenraj could you please use the decorator `@with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")` first before the other decorators? Thanks!

@RickSanchezStoic The decorator @with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle") is already added. Apologies for the confusion caused by the commits.  Could you please assist me with the failing tests? Your guidance would be greatly appreciated. Thank you.

**Comment by RickSanchezStoic** [comment]
Hi @remenraj could you please update your repo once .

**Comment by remenraj** [comment]
Hi @RickSanchezStoic, I have now updated my repository to sync with the latest changes from the original repo. Still some tests are failing though. Could you please guide me on what to do next? Thank you for your support :)

File: ivy/functional/frontends/paddle/tensor/math.py [hunk 1]
```diff
@@ -319,6 +319,12 @@ def trunc(x, name=None):
     return ivy.trunc(x)
 
 
+@with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def digamma(x, name=None):
+    return ivy.digamma(x)
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/paddle/tensor/math.py:325]: This function isn't implemented on the functional API you'll need to write a composition🙂

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @remenraj

Thanks for the PR 😊
Kindly check the comments below for the changes needed

File: ivy/functional/frontends/paddle/tensor/math.py (hunk 1 above)
Comment (remenraj) [reply ivy/functional/frontends/paddle/tensor/math.py:325]: Hi @hirwa-nshuti. After investigating further, I found that Ivy currently lacks the gamma function and its derivative, which are required for the digamma function. I would appreciate your guidance on how to proceed with implementing the digamma function in Ivy. Thank you for your support.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**